### PR TITLE
New --log-level option.

### PR DESCRIPTION
**Pull request by mnottale**
Signed-off-by: Matthieu Nottale <matthieu.nottale@docker.com>

### CODE REVIEW

File: compose/cli/main.py [hunk 1]
```diff
@@ -139,18 +139,35 @@ def setup_parallel_logger(noansi):
...
+          'DEBUG': logging.DEBUG ,
+          'INFO': logging.INFO,
+          'WARNING': logging.WARNING,
+          'ERROR': logging.ERROR,
+          'CRITICAL': logging.CRITICAL,
+        }
+        loglevel = levels.get(level)
+        if loglevel is None:
+            try:
+                loglevel = int(level)
```
Comment (shin-) [inline compose/cli/main.py]: I think it'll be easier to document and easier to use if we only support string values.

File: compose/cli/main.py [hunk 2]
```diff
@@ -139,18 +139,35 @@ def setup_parallel_logger(noansi):
...
+
+    if level is not None:
+        levels = {
+          'DEBUG': logging.DEBUG ,
+          'INFO': logging.INFO,
+          'WARNING': logging.WARNING,
+          'ERROR': logging.ERROR,
+          'CRITICAL': logging.CRITICAL,
+        }
+        loglevel = levels.get(level)
```
Comment (shin-) [inline compose/cli/main.py]: maybe we can do `level.upper()` here - not much reason to be case-sensitive.

**Comment by shin-** [comment]
Fixes #5590 
Fixes #3447

File: compose/cli/main.py [hunk 3]
```diff
@@ -100,7 +100,10 @@ def dispatch():
         {'options_first': True, 'version': get_version_info('compose')})
 
     options, handler, command_options = dispatcher.parse(sys.argv[1:])
-    setup_console_handler(console_handler, options.get('--verbose'), options.get('--no-ansi'))
+    setup_console_handler(console_handler,
+                          options.get('--verbose'),
+                          options.get('--no-ansi'),
+                          options.get("--log-level"))
```
Comment (ebuildy) [inline compose/cli/main.py]: What about adding also an env. variable here? (`COMPOSE_LOG_LEVEL`)

**Pull request merged**